libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -493,7 +493,7 @@ - (NSString *)devInfoTitleString
                 [presentedViewController dismissViewControllerAnimated:YES completion:^{
                     // Restart authentication with the updated configuration
                     if ([presentedViewController isKindOfClass:[SFLoginViewController class]]) {
-                        [[SFUserAccountManager sharedInstance] restartAuthenticationForViewController:(SFLoginViewController *)presentedViewController];
+                        [[SFUserAccountManager sharedInstance] restartAuthenticationForViewController:(SFLoginViewController *)presentedViewController recreateAuthRequest:YES];
```

Comment:
Login options might change the consumer key / callback url / scopes so we need to restart the oauth request otherwise the new settings will not be used.